src/creature_tracker.cpp

```diff
@@ -161,6 +204,12 @@ void Creature_tracker::remove( const int idx )
             --elem.second;
         }
     }
+
+    for( auto &elem : monsters_by_id ) {
+        if( elem.second > ( size_t )idx ) {
+            --elem.second;
```

Comment:
Holy crap this is fragile, this seriously needs an overhaul.
If nothing else monster_list should be a list, and then the other two can have pointers to the monsters.